packages/plugin-data/src/api/data/soql/reporters.ts

```diff
@@ -0,0 +1,326 @@
+
+  public constructor(ux: UX, logger: Logger) {
+    this.ux = ux;
+    this.logger = logger.child('reporter');
+  }
+
+  public log(...args: string[]): void {
+    this.ux.log(...args);
+  }
+}
+
+export class QueryReporter extends Reporter {
+  protected columns: Field[] = [];
+  protected data: DataSoqlQueryResult;
+
+  public constructor(data: DataSoqlQueryResult, columns: Field[], ux: UX, logger: Logger) {
+    super(ux, logger);
+    this.columns = columns;
+    this.data = data;
+  }
+}
+
+type ParsedFields = {
+  attributeNames: string[];
+  children: string[];
+  aggregates: FunctionField[];
+};
+
+type ColumnAttributes = {
+  key: string;
+  label: string;
+};
+
+export class HumanReporter extends QueryReporter {
+  public constructor(data: DataSoqlQueryResult, columns: Field[], ux: UX, logger: Logger) {
+    super(data, columns, ux, logger);
+  }
+
+  public display(): void {
+    const { attributeNames, children, aggregates } = this.parseFields();
+    const totalCount = this.data.result.records.length;
+    this.soqlQuery(attributeNames, this.massageRows(this.data.result.records, children, aggregates), totalCount);
+  }
+
+  public parseFields(): ParsedFields {
+    const fields = this.columns;
+    // Field names
+    const attributeNames: string[] = [];
+
+    // For subqueries. Display the children under the parents
+    const children: string[] = [];
+
+    // For function fields, like avg(total).
+    const aggregates: FunctionField[] = [];
+
+    if (fields) {
+      this.logger.info(`Found fields ${JSON.stringify(fields.map((field) => `${typeof field}.${field.name}`))}`);
+
+      fields.forEach((field) => {
+        if (field instanceof SubqueryField) {
+          children.push(field.name);
+          field.fields
+            .map((subfield: Field) => subfield as SubqueryField)
+            .forEach((subfield: SubqueryField) => attributeNames.push(`${field.name}.${subfield.name}`));
+        } else if (field instanceof FunctionField) {
+          if (field.alias) {
+            attributeNames.push(field.alias);
+          } else {
+            attributeNames.push(field.name);
+          }
+          aggregates.push(field);
+        } else {
+          attributeNames.push(field.name);
+        }
+      });
+    } else {
+      this.logger.info(`No fields found for query "${this.data.query}"`);
+    }
+
+    return { attributeNames, children, aggregates };
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public soqlQuery(columns: string[], records: any[], totalCount: number): void {
+    this.prepNullValues(records);
+    this.ux.table(records, { columns: this.prepColumns(columns) });
+    this.log(chalk.bold(messages.getMessage('displayQueryRecordsRetrieved', [totalCount])));
+  }
+
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  public prepNullValues(records: any[]): void {
+    records.forEach((record): void => {
+      Reflect.ownKeys(record).forEach((propertyKey) => {
```

Comment:
Where is `Reflect` coming from? If it is a library can we get workaround using it?

Reply:
@WillieRuemmele [Reflect](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect) is part of JS and I prefer it over using Objects. What I would look for to replace it would be equivalent functions from our own libraries.

I will look for suitable replacements.

Reply:
`Reflect` is new to me, but it looks handy.

Reply:
This is approaching personal preference so not sure it's worth spending much time discussing it, but we might want to bring it up as a team during planning or a parking lot, at least so that everyone on the team is aware.  There's a balance between code consistency for easier maintenance and freedom to write code to your style.  This is really only relevant when object properties are symbols or non-enumerable.  Maybe we should add something to the `kit` library for this if there isn't something there already?

Reply:
@shetzel, I am all for providing this functionality in kit.